src/metabase/lib/join.cljc

```diff
@@ -620,10 +621,15 @@
    ;; I was on the fence about whether these should get `:lib/source :source/joins` or not -- it seems like based on
    ;; the QB UI they shouldn't. See screenshots in #31174
    (sort-join-condition-columns
-    (let [joinable (if (join? join-or-joinable)
+    (let [joinable   (if (join? join-or-joinable)
```

Comment:
`join-condition-rhs-columns` should include the join alias if we're using it with an existing join; and it should include `:source/joins` so we trigger the join-group logic in `lib.column-group`